### ISSUE TITLE
An exception name occurs when parsing dbc in canFD format

### ISSUE THREAD

**Issue by MagicKim** [xR3b0rn/dbcppp#105]
1.use command ：dbcppp dbc2 --dbc=create_canfd.dbc --format=C
2.my test dbc file：
```
VERSION "created by canmatrix"

NS_ :

BS_:

BU_:


BO_ 2147483651 CanExtended3: 8 Vector__XXX

BO_ 4 CanStandard4: 8 Vector__XXX

BO_ 2147483650 CanFdExtended2: 16 Vector__XXX
 SG_ signal2 : 64|64@1- (1,0) [0|0] "" Vector__XXX
 SG_ signal1 : 0|64@1- (1,0) [0|0] "" Vector__XXX

BO_ 1 canFdStandard1: 24 Vector__XXX
 SG_ signal3 : 128|64@1- (1,0) [0|0] "" Vector__XXX
 SG_ signal2 : 64|64@1- (1,0) [0|0] "" Vector__XXX
 SG_ signal1 : 0|64@1- (1,0) [0|0] "" Vector__XXX

BA_DEF_ "BusType" STRING;
```
3.parser output C:
```
#ifndef DBCPPP_BYTE_ORDER_LITTLE_ENDIAN
#error "Please pass -DDBCPPP_BYTE_ORDER_LITTLE_ENDIAN=<1|0>"
#endif
#include <stdint.h>
#define bswap_16(value) ((((value) & 0xff) << 8) | ((value) >> 8))
#define bswap_32(value) \
    (((uint32_t)bswap_16((uint16_t)((value) & 0xffff)) << 16) | \
    (uint32_t)bswap_16((uint16_t)((value) >> 16)))
#define bswap_64(value) \
    (((uint64_t)bswap_32((uint32_t)((value) & 0xffffffff)) \
    << 32) | \
    (uint64_t)bswap_32((uint32_t)((value) >> 32)))
uint64_t dbcppp_native_to_big(uint64_t& v)
{
#if DBCPPP_BYTE_ORDER_LITTLE_ENDIAN == 1
    return bswap_64(v);
#else
    return v;
#endif
}
uint64_t dbcppp_native_to_little(uint64_t& v)
{
#if DBCPPP_BYTE_ORDER_LITTLE_ENDIAN == 1
    return v;
#else
    return bswap_64(v);
#endif
}
uint64_t dbcppp_decode_CanFdExtended2_2147483650_signal2(const void* nbytes)
{
    uint64_t data;
    hack.ui = *reinterpret_cast<const uint64_t*>(&reinterpret_cast<const uint8_t*>(nbytes)[8]);
    uint64_t data1 = reinterpret_cast<const uint8_t*>(nbytes)[8 + 8];
    data = dbcppp_native_to_big(data);
    data &= 144115188075855871ull;
    data <<= 7ull;
    data1 >>= 1ull;
    data |= data1;
    return data;
}
double dbcppp_rawToPhys_CanFdExtended2_2147483650_signal2(uint64_t value)
{
    return value * 1 + 0;
}
uint64_t dbcppp_decode_CanFdExtended2_2147483650_signal1(const void* nbytes)
{
    uint64_t data;
    hack.ui = *reinterpret_cast<const uint64_t*>(&reinterpret_cast<const uint8_t*>(nbytes)[0]);
    uint64_t data1 = reinterpret_cast<const uint8_t*>(nbytes)[0 + 8];
    data = dbcppp_native_to_big(data);
    data &= 144115188075855871ull;
    data <<= 7ull;
    data1 >>= 1ull;
    data |= data1;
    return data;
}
double dbcppp_rawToPhys_CanFdExtended2_2147483650_signal1(uint64_t value)
{
    return value * 1 + 0;
}
uint64_t dbcppp_decode_canFdStandard1_1_signal3(const void* nbytes)
{
    uint64_t data;
    hack.ui = *reinterpret_cast<const uint64_t*>(&reinterpret_cast<const uint8_t*>(nbytes)[16]);
    uint64_t data1 = reinterpret_cast<const uint8_t*>(nbytes)[16 + 8];
    data = dbcppp_native_to_big(data);
    data &= 144115188075855871ull;
    data <<= 7ull;
    data1 >>= 1ull;
    data |= data1;
    return data;
}
double dbcppp_rawToPhys_canFdStandard1_1_signal3(uint64_t value)
{
    return value * 1 + 0;
}
uint64_t dbcppp_decode_canFdStandard1_1_signal2(const void* nbytes)
{
    uint64_t data;
    hack.ui = *reinterpret_cast<const uint64_t*>(&reinterpret_cast<const uint8_t*>(nbytes)[8]);
    uint64_t data1 = reinterpret_cast<const uint8_t*>(nbytes)[8 + 8];
    data = dbcppp_native_to_big(data);
    data &= 144115188075855871ull;
    data <<= 7ull;
    data1 >>= 1ull;
    data |= data1;
    return data;
}
double dbcppp_rawToPhys_canFdStandard1_1_signal2(uint64_t value)
{
    return value * 1 + 0;
}
uint64_t dbcppp_decode_canFdStandard1_1_signal1(const void* nbytes)
{
    uint64_t data;
    hack.ui = *reinterpret_cast<const uint64_t*>(&reinterpret_cast<const uint8_t*>(nbytes)[0]);
    uint64_t data1 = reinterpret_cast<const uint8_t*>(nbytes)[0 + 8];
    data = dbcppp_native_to_big(data);
    data &= 144115188075855871ull;
    data <<= 7ull;
    data1 >>= 1ull;
    data |= data1;
    return data;
}
double dbcppp_rawToPhys_canFdStandard1_1_signal1(uint64_t value)
{
    return value * 1 + 0;
}

``` 
problem: 
look parser C code (dbcppp_rawToPhys_canFdStandard1_1_signal2)  function name "hack.ui"???

**Comment by xR3b0rn** [comment]
What is the exact problem?

**Comment by MagicKim** [comment]
```
1.uint64_t dbcppp_decode_CanFdExtended2_2147483650_signal2(const void* nbytes)
2.{
3.    uint64_t data;
4.    hack.ui = *reinterpret_cast<const uint64_t*>(&reinterpret_cast<const uint8_t*>(nbytes)[8]);
5.    uint64_t data1 = reinterpret_cast<const uint8_t*>(nbytes)[8 + 8];
6.    data = dbcppp_native_to_big(data);
7.    data &= 144115188075855871ull;
8.    data <<= 7ull;
9.    data1 >>= 1ull;
10.    data |= data1;
11.    return data;
12.}
```
The variable name on the fourth line is invalid。 **hack.ui**

**Comment by xR3b0rn** [comment]
Fixed in https://github.com/xR3b0rn/dbcppp/commit/067abf5609788b24626ab9b496aa96d4b5d252a5

**Comment by MagicKim** [comment]
> Fixed in [067abf5](https://github.com/xR3b0rn/dbcppp/commit/067abf5609788b24626ab9b496aa96d4b5d252a5) You can pull from the master.

thank you very much
